Adding Log Info Documentation (#305)
lhr-solar/BPS: Docs/source/Software/Tasks.rst

Additional Considerations
    When modifying the idle task, it is important to not introduce any functionality that may affect other tasks. For example, the idle task should not pend 
    any mutexes, since this could block more important tasks from running.
                   
Log Info Task
=============

Purpose

>    This task logs the state of charge into EEPROM every 3 seconds.

Reviewer: ```suggestion
   This task logs the state of charge into the EEPROM every 3 seconds.
```

Functionality

>    The log info task runs an infinite loop and is delayed by OSTimeDly every 3 seconds. 

Reviewer: What else does it do inside of the infinite loop?

Priority

>    This task has priority 8, so it will not interrupt any monitoring tasks or any tasks that check if the BPS is running correctly.

Reviewer: It also will have a lower priority than sending CAN messages

Shared Resources

>    The log info task shares battery charge data and the EEPROM, which is also shared by the Fault State Task and CLI. 

Reviewer: ```suggestion
   The log info task uses battery state of charge data and the EEPROM, which is also shared by the Fault State Task and CLI. 
```

Timing Requirements

>    There is a time requirement of logging into the EEPROM every 3 seconds. Writing too often to the EEPROM will exceed the EEPROM's limited erase/write         cycles, which causes the EEPROM to malfunction. Thus, writing every 3 secconds will update the EEPROM accurately enough and stay within the EEPROM's         erase/write cycles.   

Reviewer: ```suggestion
    There is a time requirement of logging into the EEPROM every 3 seconds. Writing too often to the EEPROM will exceed the EEPROM's limited (4 million) erase/write
    cycles, which causes the EEPROM to malfunction. Thus, writing every 3 seconds will update the state of charge accurately enough and stay within the EEPROM's
    erase/write cycles.   
```

Additional Considerations
    When modifying the idle task, it is important to not introduce any functionality that may affect other tasks. For example, the idle task should not pend 
    any mutexes, since this could block more important tasks from running.
                   
Log Info Task
=============

Purpose
   This task logs the state of charge into the EEPROM every 3 seconds.

Functionality

>    The log info task runs an infinite loop. Inside the loop, it sends percentage of charge left in the battery pack to the EEPROM using EEPROM_LogData and is   then delayed by OSTimeDly every 3 seconds. 

Reviewer: ```suggestion
   The log info task runs an infinite loop. Inside the loop, it sends percentage of charge left in the battery pack to the EEPROM using ``EEPROM_LogData()`` and is then delayed by ``OSTimeDly()`` every 3 seconds. 
```

Everything enclosed in the double backticks (``) gets formatted as code instead of regular text

Priority

>    This task has priority 8, so it will not interrupt any monitoring tasks or any tasks that check if the BPS is running correctly. It will also have a lower   priority than sending CAN messages.

Reviewer: ```suggestion
   This task has priority 8, so it will not interrupt any monitoring tasks or any tasks that check if the BPS is running correctly. It will also have a lower priority than sending CAN messages.
```

Shared Resources
   The log info task uses battery state of charge data and the EEPROM, which is also shared by the Fault State Task and CLI. 

Timing Requirements
   There is a time requirement of logging into the EEPROM every 3 seconds. Writing too often to the EEPROM will exceed the EEPROM's limited (4 million) 

>    erase/write cycles, which causes the EEPROM to malfunction. Thus, writing every 3 secconds will update the EEPROM accurately enough and stay within the

Reviewer: ```suggestion
   erase/write cycles, which causes the EEPROM to malfunction. Thus, writing every 3 seconds will update the EEPROM accurately enough and stay within the
```

   EEPROM's erase/write cycles.   
 
Yields

>    The log info task yields when OSTimeDly is called and when EEPROM is initialized and written to. 

Reviewer: ```suggestion
   The log info task yields when ``OSTimeDly()`` is called and when the EEPROM is initialized and written to. 
```
